apps/server/src/modules/files-storage/uc/files-storage-download.uc.spec.ts

```diff
@@ -0,0 +1,281 @@
+		authorizationService = module.get(AuthorizationService);
+		filesStorageService = module.get(FilesStorageService);
+	});
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+	});
+
+	afterAll(async () => {
+		await orm.close();
+		await module.close();
+	});
+
+	it('should be defined', () => {
+		expect(filesStorageUC).toBeDefined();
+	});
+
+	describe('download is called', () => {
+		describe('WHEN file is found and user is authorized', () => {
+			const setup = () => {
+				const { fileRecord, params, userId } = getFileRecordWithParams();
+				const fileDownloadParams = { ...params, fileName: fileRecord.name };
+
+				filesStorageService.getFileRecord.mockResolvedValueOnce(fileRecord);
+
+				return { fileDownloadParams, userId, fileRecord };
+			};
+
+			it('should call getFile with correct params', async () => {
+				const { fileDownloadParams, userId } = setup();
+
+				await filesStorageUC.download(userId, fileDownloadParams);
+
+				expect(filesStorageService.getFileRecord).toHaveBeenCalledWith({
+					fileRecordId: fileDownloadParams.fileRecordId,
+				});
+			});
+
+			it('should call checkPermissionByReferences with correct params', async () => {
+				const { fileDownloadParams, userId, fileRecord } = setup();
+
+				await filesStorageUC.download(userId, fileDownloadParams);
+
+				const allowedType = FilesStorageMapper.mapToAllowedAuthorizationEntityType(fileRecord.parentType);
+				expect(authorizationService.checkPermissionByReferences).toHaveBeenCalledWith(
+					userId,
+					allowedType,
+					fileRecord.parentId,
+					PermissionContexts.read
+				);
+			});
+		});
+
+		describe('WHEN getFile throws error', () => {
+			const setup = () => {
+				const { fileRecord, params, userId } = getFileRecordWithParams();
+				const fileDownloadParams = { ...params, fileName: fileRecord.name };
+				const error = new Error('test');
+
+				filesStorageService.getFileRecord.mockRejectedValueOnce(error);
+
+				return { fileDownloadParams, userId, error };
+			};
+
+			it('should pass error', async () => {
+				const { fileDownloadParams, error, userId } = setup();
+
+				await expect(filesStorageUC.download(userId, fileDownloadParams)).rejects.toThrowError(error);
+			});
+		});
+
+		describe('WHEN user is not authorized', () => {
+			const setup = () => {
+				const { fileRecord, params, userId } = getFileRecordWithParams();
+				const fileDownloadParams = { ...params, fileName: fileRecord.name };
+				const error = new ForbiddenException();
+
+				filesStorageService.getFileRecord.mockResolvedValueOnce(fileRecord);
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(error);
+
+				return { fileDownloadParams, userId, fileRecord };
+			};
+
+			it('should throw Error', async () => {
+				const { fileDownloadParams, userId } = setup();
+
+				await expect(filesStorageUC.download(userId, fileDownloadParams)).rejects.toThrow();
+			});
+		});
+
+		describe('WHEN file is successfully downloaded', () => {
```

Comment:
WHEN file is found and user is authorized  vs WHEN file is successfully downloaded